
syntax = "proto3";

package PGo;

message FortDetailsOutProto {
  string id = 1;
  string name = 4;
  string image_url = 5;
  bool isPokeStop = 9;
  double lat = 10;
  double long = 11;
  string description = 12;
  message Lure {
    uint32 something = 1;
    uint64 expiration = 2;
    string trainer = 3;
  }
  Lure lure = 13;
}

message FortDetailsProto {
  string id = 1;
  double lat = 2;
  double long = 3;
}

message FortSearchProto {
  string id = 1;

  //Trainer and Fort for calculating distance?
  double lat = 2;
  double long = 3;
  double lat2 = 4;
  double long2 = 5;
}

//I'm making this up as a I go along
message FortSearchOutProto {
  bool success = 1;
  message Result {
    uint64 ts1 = 1;
    uint64 ts2 = 2;
    repeated Item items = 3;
  }
  Result result = 2;
}

message Item {
  uint64 timestamp = 1;
  //Stuff nested under '3'
}


message Gym {
  message Details {
    message MoreDetails {
      string id = 1;
      uint64 timestamp = 2;
      double lat = 3;
      double long = 4;
      //5, 6, 8
      uint64 prestige = 10;
    }
    message Level {
      message Pokemon {
        uint64 cp = 3;
        uint64 currentHealth = 4;
        uint64 maxHealth = 5;
        uint64 moveOne = 6;
        uint64 moveTwo = 7;
        string trainer = 9;
        float height = 15;
        float weight = 16;
        uint64 CapturedS2CellId = 22;
        uint64 CreationTimeMs = 26;
        string nickname = 30;
      }

      message Trainer {
        string name = 1;
        uint64 level = 2;
      }

      Pokemon pokemon = 1;
      Trainer trainer = 2;
    }

    MoreDetails details = 1;
    repeated Level levels = 2;
  }

  Details details = 1;
  string name = 2;
  string image_url = 3;
  uint64 team = 4;
  string description = 5;
}
